archive/test_archive.py

```diff
@@ -0,0 +1,215 @@
+import os
+
+import unittest
+from unittest.mock import MagicMock, patch, call
+from pymssql import exceptions
+from archive import (
+    get_connection,
+    get_all_plant_ids,
+    get_plants_data,
+    upload_plant_metric_data,
+    calculate_archive_metrics,
+    get_latest_recording,
+    clear_plant_metrics,
+    lambda_handler
+)
+
+
+class TestArchive(unittest.TestCase):
+    """ Test class containing archive tests """
+    @patch.dict(
+        os.environ,
+        {
+            "DB_HOST": "mock_value",
+            "DB_NAME": "mock_value",
+            "DB_USER": "mock_value",
+            "DB_PASSWORD": "mock_value",
+            "DB_PORT": "mock_value",
+            "SERVER": "mock_value"
+        },
+    )
+    @patch('archive.connect')
+    def test_get_connection_success(self, mock_connect):
+        """Test successful database connection."""
+        mock_connection = MagicMock()
+        mock_connect.return_value = mock_connection
+
+        connection = get_connection()
+        mock_connect.assert_called_once_with(
+            server='mock_value',
+            port='mock_value',
+            user='mock_value',
+            password='mock_value',
+            database='mock_value',
+            as_dict=True
+        )
+        assert connection == mock_connect.return_value
+
+    @patch("archive.connect")
+    def test_get_connection_missing_env_var(self, mock_connect):
+        """"""
+        with self.assertRaises(KeyError):
+            get_connection()
+
+    @patch.dict(
+        os.environ,
+        {
+            "DB_HOST": "mock_value",
+            "DB_NAME": "mock_value",
+            "DB_USER": "mock_value",
+            "DB_PASSWORD": "mock_value",
+            "DB_PORT": "mock_value",
+        },
+    )
```

Comment:
seems a bit random, move to top of class please

Reply:
no!

Reply:
ITS NOT A FIXTURE